### PR TITLE
[ROCKETMQ-203]Support client to allocate message queue in machine room nearby priority

### PR DESCRIPTION
**Pull request by Jaskey**
JIRA: https://issues.apache.org/jira/browse/ROCKETMQ-203?jql=project%20%3D%20ROCKETMQ

### CODE REVIEW

File: client/src/main/java/org/apache/rocketmq/client/consumer/rebalance/AllocateMachineRoomNearby.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+        return allocateResults;
+    }
+
+    @Override
+    public String getName() {
+        return "MACHINE_ROOM_NEARBY" + "-" + allocateMessageQueueStrategy.getName();
+    }
+
+    public interface MachineRoomSelector {
```
Comment (lizhanhui) [inline client/src/main/java/org/apache/rocketmq/client/consumer/rebalance/AllocateMachineRoomNearby.java]: How about rename this interface as `MachineRoomResolver`?

Comment (Jaskey) [inline client/src/main/java/org/apache/rocketmq/client/consumer/rebalance/AllocateMachineRoomNearby.java]: LGTM，I will rename this interface if more guys agree

**Comment by Jaskey** [comment]
@lizhanhui @dongeforever @shroman 
What's your advice on this issue and implementation?

**Comment by lizhanhui** [comment]
@dongeforever @shroman @zhouxinyu Please review.

**Pull request merged**